src/platform/intel/cavs/lib/clk.c

```diff
@@ -20,11 +20,30 @@ static int clock_platform_set_cpu_freq(int clock, int freq_idx)
 	/* set CPU frequency request for CCU */
 #if CAVS_VERSION == CAVS_VERSION_1_5
 	io_reg_update_bits(SHIM_BASE + SHIM_CLKCTL, SHIM_CLKCTL_HDCS, 0);
-#endif
 	io_reg_update_bits(SHIM_BASE + SHIM_CLKCTL,
 			   SHIM_CLKCTL_DPCS_MASK(cpu_get_id()),
 			   enc);
+#else
+	uint32_t status_mask = cpu_freq_status_mask[freq_idx];
+
+	/* request clock */
+	io_reg_write(SHIM_BASE + SHIM_CLKCTL,
+		     io_reg_read(SHIM_BASE + SHIM_CLKCTL) | enc);
```

Comment:
@jajanusz shouldn't shim_read*() / shim_write*() be used for these?

Reply:
Yea it'd be better to use them. For read and write, however there is no shim_update what would mean even more read/write, but I may change it to use what you proposed.

Reply:
@jajanusz This could be a separate PR though as existing code uses io_reg_ and friends.

Reply:
@kv2019i yea i think so, also these shim_ methods are repeated like 4 times, it'd be better to just move them to common cavs headers, add utility update functions then use them here

Reply:
@lyakh are you ok with that decision?